internal/controller/pdbwatcher_controller.go

```diff
@@ -3,311 +3,117 @@ package controllers
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      pdb.Name,
+				Namespace: pdb.Namespace,
+			},
+			Spec: PDBWatcherSpec{
+				Name: pdb.Name,
+			},
+			Status: PDBWatcherStatus{
+				Status: "created",
+			},
 		}
-	}
-	return nil
-}
-
-func (r *PDBWatcherReconciler) discoverDeployment(ctx context.Context, pdb *policyv1.PodDisruptionBudget) (string, error) {
-	logger := log.FromContext(ctx)
-	// Check if PDB overlaps with multiple deployments
-	selector, err := metav1.LabelSelectorAsSelector(pdb.Spec.Selector)
-	if err != nil {
-		return "", err // Error converting label selector
-	}
-
-	podList := &corev1.PodList{}
-	err = r.List(ctx, podList, &client.ListOptions{Namespace: pdb.Namespace, LabelSelector: selector, Limit: 1})
-	if err != nil {
-		return "", err // Error listing pods
-	}
 
-	deployments := []string{}
-	for _, pod := range podList.Items {
-		for _, ownerRef := range pod.OwnerReferences {
-			if ownerRef.Kind == "ReplicaSet" {
-				replicaSet := &appsv1.ReplicaSet{}
-				err = r.Get(ctx, types.NamespacedName{Name: ownerRef.Name, Namespace: pdb.Namespace}, replicaSet)
-				if err != nil {
-					return "", err // Error fetching ReplicaSet
-				}
-
-				// Get the Deployment that owns this ReplicaSet
-				for _, rsOwnerRef := range replicaSet.OwnerReferences {
-					if rsOwnerRef.Kind == "Deployment" {
-						deployments = append(deployments, rsOwnerRef.Name)
-					}
-				}
-			}
-			//todo handle stateful sets
+		err := r.Create(ctx, &pdbWatcher)
+		if err != nil {
+			return reconcile.Result{}, fmt.Errorf("unable to create PDBWatcher: %v", err)
 		}
+		log.Log.Info("Created PDBWatcher", "name", pdb.Name)
 	}
 
-	// If multiple deployments are found, log a warning and return an error
-	if len(deployments) > 1 {
-		r.Recorder.Event(pdb, corev1.EventTypeWarning, "MultipleDeployments", "PDB overlaps with multiple deployments") //should we event on pdb watcher?
-		return "", fmt.Errorf("PDB %s/%s overlaps with multiple deployments", pdb.Namespace, pdb.Name)
-	}
-
-	if len(deployments) == 0 {
-		return "", fmt.Errorf("PDB %s/%s overlaps with zero deployments", pdb.Namespace, pdb.Name)
-	}
-	// Log the deployment map and deployment name for debugging
-	logger.Info(fmt.Sprintf("Determined Deployment name: %s->%s", pdb.Name, deployments[0]))
-	return deployments[0], nil
+	// Return no error and no requeue
+	return reconcile.Result{}, nil
 }
 
+// SetupWithManager sets up the controller with the Manager.
 func (r *PDBWatcherReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// Set up the controller to watch Deployments and trigger the reconcile function
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&myappsv1.PDBWatcher{}).
+		For(&policyv1.PodDisruptionBudget{}).
 		WithEventFilter(predicate.Funcs{
-			// ignore status updates as we make those.
-			UpdateFunc: func(ue event.UpdateEvent) bool {
-				return ue.ObjectOld.GetGeneration() != ue.ObjectNew.GetGeneration()
+			// Only trigger for Create and Delete events
```

Comment:
theoretically you could have a pdb update and change its label selectors in which case you might need to update the deployment target?

Reply:
thinking about how to solve this issue